app/src/main/java/edu/nextstep/camp/calculator/MainContract.kt

```diff
@@ -0,0 +1,22 @@
+package edu.nextstep.camp.calculator
+
+import edu.nextstep.camp.calculator.domain.Calculator
+import edu.nextstep.camp.calculator.domain.Expression
+import edu.nextstep.camp.calculator.domain.Operator
+
+interface MainContract {
+    interface View {
+        fun showExpression(result: String)
+        fun showToast(message: String)
+    }
+
+    interface Presenter {
+        var expression : Expression
+        var operator : Operator
+        var calculator : Calculator
+        fun clickOperator(operator: Operator)
+        fun clickOperand(operand : Int)
+        fun clickEqual()
+        fun clickDelete()
```

Comment:
Presenter의 인터페이스 함수명은 어떤 의미로 지으면 좋을까를 예전에 MVP를 구현할 당시 고민했던 부분입니다.
비지니스적인 관점에서 Presenter에 요청하는 Action 형태로 선언할지, View의 관점에서 User Event 형태로 선언할지 고민을 많이 했었고, 아키텍처는 MVVM까지 변해왔지만 대부분 전자 형태로 많이 선언합니다.

```suggestion
fun addToExpression(operator: Operator)
fun addToExpression(operand : Int)
fun calculateExpression()
fun removeLastFromExpression()
```

이렇게 바꿔 보시면 어떨까요? 개인적으로는 Presenter API 명은 로직을 표현하는게 좋다는 생각입니다. 🙏
https://medium.com/@cervonefrancesco/model-view-presenter-android-guidelines-94970b430ddf 
이글을 한번 읽어보시는것도 도움이 될것 같습니다.

Reply:
말씀대로 presenter의 역할이 view에 연결뿐만아니라 model과의 연결도 있기 때문에 click~은 한정적인 것 같네요!